sense.go

```diff
@@ -0,0 +1,220 @@
+package rpmpack
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type rpmSense uint32
+
+// SenseAny specifies no specific version compare
+// SenseLess specifies less then the specified version
+// SenseGreater specifies greater then the specified version
+// SenseEqual specifies equal to the specified version
+const (
+	SenseAny  rpmSense = 0
+	SenseLess rpmSense = 1 << iota
+	SenseGreater
+	SenseEqual
+)
+
+type relationCategory string
+
+const (
+	RequiresCategory   relationCategory = "requires"
+	ObsoletesCategory  relationCategory = "obsoletes"
+	SuggestsCategory   relationCategory = "suggests"
+	RecommendsCategory relationCategory = "recommends"
+	ConflictsCategory  relationCategory = "conflicts"
+	ProvidesCategory   relationCategory = "provides"
+)
+
+var relationMatch = regexp.MustCompile(`([^=<>\s]*)\s*((?:=|>|<|>=|<=)*)\s*(.*)?`)
```

Comment:
I think the inner capture group should have ? and not *. Instead of `((?:=|>|<|>=|<=)*)` you want `((?:=|>|<|>=|<=)?)`

Because it is 0 or 1 cases of relation operators, not 0 or more (i.e. we don'
t want `===` to be accepted)

Reply:
this is handled by the parseSense returning an error. You need to capture all the `=|>|<` otherwise they become part of the version on the regex grouping, or if you ensure to exclude those from the version part it will just drop the version with no error.

https://play.golang.org/p/mKixrlOoHvX